From 48bc2e3b5f66ff7d623dc6acd4201d1c18ddee68 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 3 May 2024 11:49:01 +0200
Subject: [PATCH 1122/1141] irqchip: rpmsg: fix may be used uninitialized
 warning

The 'ret' variable is used uninitialized in dev_err.
Store in 'ret' variable the irq_rpmsg_read_val() return to fix
the issue.

Change-Id: Ia2807f4b07ccafa9dc28bacce6da6606d8805ef9
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/378198
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/irqchip/irq-rpmsg.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-rpmsg.c b/drivers/irqchip/irq-rpmsg.c
index f84e8c163fa3..f0f15ccb9ff9 100644
--- a/drivers/irqchip/irq-rpmsg.c
+++ b/drivers/irqchip/irq-rpmsg.c
@@ -164,7 +164,8 @@ static void irq_rpmsg_setup_work(struct work_struct *ws)
 	struct rpmsg_intc_msg msg = {VIRT_INTC_CFG, 0, 0};
 	int ret;
 
-	if (irq_rpmsg_read_val(rirq_dev, &msg)) {
+	ret = irq_rpmsg_read_val(rirq_dev, &msg);
+	if (ret) {
 		dev_err(&rpdev->dev, "read intc cfg failed: %d\n", ret);
 		goto err;
 	}
-- 
2.39.2

